libqtile/core/manager.py

```diff
@@ -352,6 +355,14 @@ def process_key_event(self, keysym: int, mask: int) -> None:
         if isinstance(key, KeyChord):
             self.grab_chord(key)
         else:
+            if key.on_release != is_release:
+                return
+            if key_as_modifier and key.on_release and last_key_mask & key_as_modifier:
+                # quick fix for on_release keys:
+                # When using a modifier key like Super_L as a regular key,
+                # it should not be triggered when released as the last key of a
+                # sequence of modifiers / keys.
+                return
```

Comment:
This solution feels like it's the business of the backend to ensure that it isn't sending the qtile manager binding events that weren't asked for by the manager. I can see that it might make sense to be here if all backends would require the same solution. I'll start exploring how the wayland backend can implement all of this to see if it makes sense to keep this here or to keep it within the x11 backend.

Reply:
One point to keep in mind is that `process_key_...` must be called even if the action is not triggered. X11 sends `KeyRelease` events for every key that is released, e.g. `Ctrl+Meta+X ~> Ctrl+Meta ~> Meta`. If `Meta` itself is bound to an on-release action, it is triggered even if I'm just releasing the modifiers of another binding I just used. To detect that, the `last_key_mask` is necessary. This makes the `process_key_event` thing a non-pure function. Another consequence in this case is that the manager needs both the symbol as well as the modmask to be X11 values.  
Not sure if that helps at all, but I still wanted to mention it.